sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -932,6 +933,33 @@ object PushPredicateThroughGenerate extends Rule[LogicalPlan] with PredicateHelp
 }
 
 /**
+ * Push [[Filter]] operators through [[Window]] operators. Parts of the predicate that can be pushed
+ * beneath must satisfy three conditions:
+ * 1. involving one and only one column that is part of window partitioning key.
```

Comment:
The same. I do not know the specific reasons. 
https://issues.apache.org/jira/secure/attachment/12788757/HIVE-12808.05.patch
In the above Hive patch, it has test case to cover it.
`Test predicate is not getting pushed down when predicate involves more than one col `